Tests/Core/TestObjectGraphContext.m

```diff
@@ -660,4 +675,94 @@ - (void) testCrossContextReferenceSerializedAsNull
     UKObjectsEqual([NSNull null], itemContentsArray[0]);
 }
 
+- (void) testCrossContextReferencedObjectGraphContextDeallocated
+{
+    @autoreleasepool {
+        COObjectGraphContext *ctx2 = [COObjectGraphContext new];
+        
+        OutlineItem *ctx2root = [[OutlineItem alloc] initWithObjectGraphContext: ctx2];
+        ctx2root.label = @"ctx2root";
+        [ctx2 setRootObject: ctx2root];
+
+        // create a link from ctx1 to ctx2
+        [root1 addObject: ctx2root];
+        
+        UKObjectsNotSame(ctx1, ctx2root.objectGraphContext);
+        UKObjectsEqual(ctx2root, root1.contents[0]);
+        
+        NSLog(@"%@", ctx1.items);
+    }
+    
+    // check that ctx1 is still valid?
+    
+    UKTrue([root1.contents isEmpty]);
+    
+    NSLog(@"%@", [ctx1 detailedDescription]);
+}
+
+- (void) testCrossContextReferencedObjectDeallocatedWithTwoReferences
+{
+    // add a child in ctx1
+    OutlineItem *ctx1obj = [[OutlineItem alloc] initWithObjectGraphContext: ctx1];
+    ctx1obj.label = @"ctx1obj";
+    
+    // create a ctx2
+    
+    COObjectGraphContext *ctx2 = [COObjectGraphContext new];
+    OutlineItem *ctx2root = [[OutlineItem alloc] initWithObjectGraphContext: ctx2];
+    ctx2root.label = @"ctx2root";
+    [ctx2 setRootObject: ctx2root];
```

Comment:
May be it's possible to replace -setRootObject: here and elsewhere in the method with .rootObject =

iirc I don't think the compiler would ask for an upcast, but I could be wrong.